cl/search/management/commands/export_doc_coverage.py

```diff
@@ -0,0 +1,155 @@
+from pathlib import Path
+
+import pandas as pd
+from django.db.models import Count, Q
+from elasticsearch_dsl import Search
+from tqdm import tqdm
+
+from cl.lib.command_utils import BaseCommand, logger
+from cl.search.documents import DocketDocument
+from cl.search.models import Docket
+
+
+class Command(BaseCommand):
+    help = """Export docket index with document coverage statistics for preparing a pretraining dataset.
+
+    Two options here, a psql and elastic approach. I'm guessing the elastic version
+    will be faster, but the psql has the advantage of pulling in stats on the number of
+    rss entries which might be useful for sampling (I couldn't figure out description == '' in es!)
+
+    Just pulls in minimal document level stats I thought might be useful for smapling a good set.
+    We can merge in other header data from the bulk exports later.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--out-path",
+            type=str,
+            required=True,
+            help="Path to save output as csv",
+        )
+        parser.add_argument(
+            "--psql",
+            action="store_true",
+            help="Use postgres approch instead of elastic",
+        )
+        parser.add_argument(
+            "--overwrite",
+            action="store_true",
+            help="Overwrite the output file if it exists",
+        )
+
+    def handle(self, *args, **options):
+        out_path = Path(options["out_path"])
+        if out_path.exists():
+            if not options["overwrite"]:
+                raise FileExistsError(
+                    f"File already exists at {options['out_path']}. Use --overwrite to overwrite."
+                )
+            out_path.unlink()
+
+        handler = self.handle_psql if options["psql"] else self.handle_es
+
+        min_year = Docket.objects.earliest("date_filed").date_filed.year
+        max_year = Docket.objects.latest("date_filed").date_filed.year
+
+        logger.info(f"Processing dockets from {min_year} to {max_year}...")
+        for year in tqdm(
```

Comment:
Can we add a configurable sleep in our for loop, perhaps? I could see this being pretty nasty on some of our biggest tables, so at least putting a sleep in it will spread the load a bit. (Should probably do it for ES too.)

Reply:
We should add a log-line too, so we know what kind of progress we're making each time we go through the loop, otherwise, we won't know if one query is taking five minutes or if the whole thing is.